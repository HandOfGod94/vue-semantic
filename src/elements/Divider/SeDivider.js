import SeDividerComponent from './SeDivider.functional'

export default {
  name: 'se-divider',
  props: {
    horizontal: Boolean,
    inverted: Boolean,
    fitted: Boolean,
    hidden: Boolean,
    section: Boolean,
    clearing: Boolean
  },

  computed: {
    classObject () {
      return {
        ui: true,
        horizontal: this.horizontal,
        inverted: this.inverted,
        fitted: this.fitted,
        hidden: this.hidden,
        section: this.section,
        clearing: this.clearing,
        divider: true
      }
    }
  },

  render (createElement) {
    return createElement(SeDividerComponent, {
      attrs: {
        slots: {...this.$slots}
      },
      class: this.classObject
    })
  }
}
