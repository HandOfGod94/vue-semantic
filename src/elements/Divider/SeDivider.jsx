import Vue from 'vue'
import Component from 'vue-class-component'

@Component({
  name: 'se-divider',
  props: {
    horizontal: Boolean,
    inverted: Boolean,
    fitted: Boolean,
    hidden: Boolean,
    section: Boolean,
    clearing: Boolean
  }
})
class SeDivider extends Vue {
  render () {
    return (
      <div class={this.classes}></div>
    )
  }

  get classes () {
    const classes = {
      ui: true,
      horizontal: this.horizontal,
      inverted: this.inverted,
      fitted: this.fitted,
      hidden: this.hidden,
      section: this.section,
      clearing: this.clearing,
      divider: true
    }

    return classes
  }
}

export default SeDivider
