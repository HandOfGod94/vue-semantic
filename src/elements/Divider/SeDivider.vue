<template>
  <div :class="classObject"></div>
</template>

<script>
export default {
  name: 'se-divider',
  props: {
    horizontal: Boolean,
    inverted: Boolean,
    fitted: Boolean,
    hidden: Boolean,
    section: Boolean,
    clearing: Boolean
  },
  computed: {
    classObject () {
      return {
        ui: true,
        horizontal: this.horizontal,
        inverted: this.inverted,
        fitted: this.fitted,
        hidden: this.hidden,
        section: this.section,
        clearing: this.clearing,
        divider: true
      }
    }
  }
}
</script>

<style>

</style>
