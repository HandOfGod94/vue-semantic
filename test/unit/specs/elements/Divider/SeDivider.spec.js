import SeDivider from '@/elements/Divider/SeDivider'
import { mount } from '@vue/test-utils'

describe('SeDivider snapshot test', () => {
  it('should match exisiting snapshot', () => {
    const wrapper = mount(SeDivider)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('SeDivider props tests', () => {
  it('should have horizontal class', () => {
    const data = { propsData: { horizontal: true } }
    const wrapper = mount(SeDivider, data)
    expect(wrapper.classes()).toContain('horizontal')
  })

  it('should have "fitted" class', () => {
    const data = { propsData: { fitted: true } }
    const wrapper = mount(SeDivider, data)
    expect(wrapper.classes()).toContain('fitted')
  })

  it('should have "clearing" class', () => {
    const data = { propsData: { clearing: true } }
    const wrapper = mount(SeDivider, data)
    expect(wrapper.classes()).toContain('clearing')
  })

  it('should have "section" class', () => {
    const data = { propsData: { section: true } }
    const wrapper = mount(SeDivider, data)
    expect(wrapper.classes()).toContain('section')
  })
})
