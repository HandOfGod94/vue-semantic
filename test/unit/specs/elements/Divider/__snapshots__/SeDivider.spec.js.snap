// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeDivider snapshot test should match exisiting snapshot 1`] = `<div class="ui divider"></div>`;
